src/careamics/models/lvae/layers.py

```diff
@@ -1333,666 +1332,4 @@ def forward(
-            The maximum value reachable by the log-variance of the latent distribution.
-            Values exceeding this threshold are clipped. Default is `None`.
-        """
-        debug_qvar_max = 0
-
-        # Check sampling options consistency
-        assert (forced_latent is None) or (not use_mode)
-
-        # Get generative distribution p(z_i|z_{i+1})
-        p_mu, p_lv, p = self.process_p_params(p_params, var_clip_max)
-        p_params = (p_mu, p_lv)
-
-        if q_params is not None:
-            # Get inference distribution q(z_i|z_{i+1})
-            # NOTE: At inference time, don't centercrop the q_params even if they are odd in size.
-            q_mu, q_lv, q = self.process_q_params(
-                q_params, var_clip_max, allow_oddsizes=mode_pred is True
-            )
-            q_params = (q_mu, q_lv)
-            sampling_distrib = q
-            debug_qvar_max = torch.max(q_lv.get())
-
-            # Centercrop p_params so that their size matches the one of q_params
-            q_size = q_mu.get().shape[-1]
-            if p_mu.get().shape[-1] != q_size and mode_pred is False:
-                p_mu.centercrop_to_size(q_size)
-                p_lv.centercrop_to_size(q_size)
-        else:
-            sampling_distrib = p
-
-        # Sample latent variable
-        z = self.get_z(
-            sampling_distrib, forced_latent, use_mode, mode_pred, use_uncond_mode
-        )
-
-        # Copy one sample (and distrib parameters) over the whole batch.
-        # This is used when doing experiment from the prior - q is not used.
-        if force_constant_output:
-            z = z[0:1].expand_as(z).clone()
-            p_params = (
-                p_params[0][0:1].expand_as(p_params[0]).clone(),
-                p_params[1][0:1].expand_as(p_params[1]).clone(),
-            )
-
-        # Pass the sampled latent througn the output convolutional layer of stochastic block
-        out = self.conv_out(z)
-
-        # Compute log p(z)# NOTE: disabling its computation.
-        # if mode_pred is False:
-        #     logprob_p =  p.log_prob(z).sum((1, 2, 3))
-        # else:
-        #     logprob_p = None
-
-        if q_params is not None:
-            # Compute log q(z)
-            logprob_q = q.log_prob(z).sum((1, 2, 3))
-            # Compute KL divergence metrics
-            kl_dict = self.compute_kl_metrics(
-                p, p_params, q, q_params, mode_pred, analytical_kl, z
-            )
-        else:
-            kl_dict = {}
-            logprob_q = None
-
-        # Store meaningful quantities to use them in following layers
-        data = kl_dict
-        data["z"] = z  # sampled variable at this layer (batch, ch, h, w)
-        data["p_params"] = p_params  # (b, ch, h, w) where b is 1 or batch size
-        data["q_params"] = q_params  # (batch, ch, h, w)
-        # data['logprob_p'] = logprob_p  # (batch, )
-        data["logprob_q"] = logprob_q  # (batch, )
-        data["qvar_max"] = debug_qvar_max
```

Comment:
Please update shapes in the comments for 3D!

Reply:
Also updated in the newer version